straxen/analyses/waveform_plot.py

```diff
@@ -338,3 +344,271 @@ def plot_wf(st: strax.Context,
+        records_matrix = False
+
+    fig = plt.figure(figsize=(25, 7 * (2 + int(records_matrix))),
+                     facecolor='white')
+    grid = plt.GridSpec((2 + int(records_matrix)), 1, hspace=0.1,
+                        height_ratios=[1.5, 0.5, 0.5][:2 + int(records_matrix)]
+                        )
+
+    # S1, S2, hitpatterns
+    gss_0 = gridspec.GridSpecFromSubplotSpec(2, 4, subplot_spec=grid[0], wspace=0.25, hspace=0.4)
+    ax_s1 = fig.add_subplot(gss_0[0])
+    ax_s2 = fig.add_subplot(gss_0[1])
+    ax_s1_hp_t = fig.add_subplot(gss_0[2])
+    ax_s1_hp_b = fig.add_subplot(gss_0[3])
+    ax_s2_hp_t = fig.add_subplot(gss_0[6])
+    ax_s2_hp_b = fig.add_subplot(gss_0[7])
+
+    # Peak & event info
+    ax_event_info = fig.add_subplot(gss_0[4])
+    ax_peak_info = fig.add_subplot(gss_0[5])
+
+    # All peaks in event
+    gss_1 = gridspec.GridSpecFromSubplotSpec(1, 1, subplot_spec=grid[1])
+    ax_ev = fig.add_subplot(gss_1[0])
+
+    # titles
+    ax_s1.set_title('Main S1')
+    ax_s1_hp_t.set_title('S1 top')
+    ax_s1_hp_b.set_title('S1 bottom')
+    ax_s2.set_title('Main S2')
+    ax_s2_hp_t.set_title('S2 top')
+    ax_s2_hp_b.set_title('S2 bottom')
+    ax_event_info.set_title('Event info')
+    ax_peak_info.set_title('Peak info')
+    ax_rec = None
+
+    # (raw)records matrix (optional)
+    if records_matrix:
+        gss_2 = gridspec.GridSpecFromSubplotSpec(1, 1, subplot_spec=grid[2])
+        ax_rec = fig.add_subplot(gss_2[0])
+
+    # Parse the hit pattern options
+    # Convert to dict (not at function definition because of mutable defaults)
+    if s1_hp_kwargs is None:
+        s1_hp_kwargs = {}
+    if s2_hp_kwargs is None:
+        s2_hp_kwargs = {}
+    # Hit patterns options:
+    for hp_opt, color_map in ((s1_hp_kwargs, "Blues"), (s2_hp_kwargs, "Greens")):
+        _common_opt = dict(xenon1t=xenon1t,
+                           pmt_label_color='lightgrey',
+                           log_scale=True,
+                           vmin=0.1,
+                           s=(250 if records_matrix else 220),
+                           pmt_label_size=7,
+                           edgecolor='grey',
+                           cmap=color_map)
+        # Is there a better way to do this update?
+        hp_opt.update(_common_opt)
+        _update = hp_opt.copy()
+        hp_opt.update(_update)
+
+    # S1
+    if events['s1_area'] > 0:
+        plt.sca(ax_s1)
+        context.plot_peaks(run_id,
+                           time_range=(events['s1_time'] - s1_fuzz,
+                                       events['s1_endtime'] + s1_fuzz),
+                           single_figure=False)
+
+        # Hit pattern plots
+        area = context.get_array(run_id, 'peaklets',
+                                 time_range=(events['s1_time'],
+                                             events['s1_endtime']),
+                                 keep_columns=('area_per_channel', 'time', 'dt', 'length'))
```

Comment:
You have now a some sort of hidden requirement. Why have you decided not to add peaks to the list of requirements for the mini-analysis?

Reply:
I decided to do it this way such that you don't have to load the peaklets for the entire event time span but only during the requested time range. Will probably not matter much computation wise but may make a slight memory difference.

Reply:
Yes this is something I have not thought of. If the user does not query any time range the memory will explode. I will also changes it in the bokeh event display.